File: frontend/src/types/document.ts
export interface ProcessedDocument {
  id: string;
  filename: string;
  content_type: string;
  file_size: number;
  status: 'uploaded' | 'processing' | 'classified' | 'validated' | 'completed' | 'success' | 'failure';
  document_type: string | null;
  classification_confidence: number | null;
  created_at: string;
  error_message: string | null;
}

export interface ProcessedDocumentDetail extends ProcessedDocument {
  extracted_data: Record<string, any> | null;
  quality_assessment: {
    quality_score: number;
    detected_issues: Array<{
      rule_name: string;
      severity: 'warning' | 'critical';
      message: string;
    }>;
    [key: string]: any;
  } | null;
  coo_insights: {
    summary: string;
    operational_action_required: boolean;
    financial_impact_estimate?: number;
    recommended_actions: string[];
    [key: string]: any;
  } | null;
}

File: backend/app/schemas/document.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional, Dict, Any, Union


class ProcessedDocumentResponse(BaseModel):
    id: UUID
    filename: str
    content_type: str
    file_size: int
    status: str
    document_type: Optional[str] = None
    classification_confidence: Optional[float] = None
    created_at: datetime
    error_message: Optional[str] = None

    class Config:
        from_attributes = True


class ProcessedDocumentDetailResponse(ProcessedDocumentResponse):
    extracted_data: Optional[Dict[str, Any]] = None
    quality_assessment: Optional[Dict[str, Any]] = None
    coo_insights: Optional[Union[Dict[str, Any], str]] = None

    class Config:
        from_attributes = True
